packages/web/src/pages/upload-page/utils/uploadPreviewContext.tsx

```diff
@@ -0,0 +1,64 @@
+import { createContext, memo, useCallback, useState } from 'react'
+
+import { Nullable, playerSelectors, queueActions } from '@audius/common'
+import { useSelector, useDispatch } from 'react-redux'
+
+const { getPlaying } = playerSelectors
+const { pause: pauseQueue } = queueActions
+
+type PreviewContextProps = {
+  playPreview: (preview: HTMLAudioElement, idx: number) => void
+  stopPreview: () => void
+  playingPreviewIndex: number
+}
+
+export const UploadPreviewContext = createContext<PreviewContextProps>({
+  playPreview: () => {},
+  stopPreview: () => {},
+  playingPreviewIndex: -1
+})
+
+export const UploadPreviewContextProvider = memo(
+  (props: { children: JSX.Element }) => {
+    const dispatch = useDispatch()
+    const isPlaying = useSelector(getPlaying)
+    const [preview, setPreview] = useState<Nullable<HTMLAudioElement>>(null)
+    const [previewIndex, setPreviewIndex] = useState<number>(-1)
+
+    const stopPreview = useCallback(() => {
+      if (preview !== null) {
+        preview.pause()
+        preview.currentTime = 0
+        setPreview(null)
+        setPreviewIndex(-1)
+      }
+    }, [preview])
+
+    const playPreview = useCallback(
+      (trackPreview: HTMLAudioElement, trackIdx: number) => {
+        if (isPlaying) dispatch(pauseQueue)
+        if (preview !== null) {
+          preview.pause()
```

Comment:
Since this stops anything playing do we need to check in the various components, or just call play preview?

Reply:
not sure what you mean?

Reply:
In the components we check if isPlayjng and then stop and then play, is that necessary if we do it here too?

Reply:
oh no, this is if something in the player is playing. the other is if a preview is playing.

Reply:
Ahh gotcha gotcha. Could we do both here?

Reply:
i guess i can condense it into a toggle function or something, but seems unnecessary when we will need the stopPreview function for other cases anyway